src/lib/state-operators/updating.ts

```diff
@@ -0,0 +1,91 @@
+import { StateOperator } from '@ngxs/store';
+import { asArray, Dictionary, mustGetActive } from '../internal';
+import { EntityStateModel } from '../models';
+import { EntitySelector, EntityUpdate, Updater } from '../actions';
+import { InvalidIdError, NoSuchEntityError, UpdateFailedError } from '../errors';
+
+export type OnUpdate<T> = (current: Readonly<T>, updated: Partial<T>) => T;
+
+/**
+ * Updates the given entities in the state.
+ * Entities will be merged with the given `onUpdate` function.
+ * @param payload the updated entities
+ * @param idKey key of the id-field of an entity
+ * @param onUpdate update function to call on each entity
+ */
+export function update<T>(
+  payload: EntityUpdate<T>,
+  idKey: string,
+  onUpdate: OnUpdate<T>
+): StateOperator<EntityStateModel<T>> {
+  return (state: EntityStateModel<T>) => {
+    let entities = { ...state.entities }; // create copy
+
+    const affected = getAffectedValues(Object.values(entities), payload.id, idKey);
```

Comment:
Shouldn't this be using the `idKey` to determine the value like this:
```suggestion
    const affected = getAffectedValues(Object.values(entities), payload[idKey], idKey);
```

Reply:
This payload is of type `EntityUpdate` and the `id` field is a `EntitySelector`.

```typescript
export  type EntitySelector<T> = string | string[] | ((entity: T) => boolean) | null;
```

I will change the field name to `selector`, as `id` is definitely misleading.